src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -509,4 +525,66 @@ public CompletableFuture<GetServerPathForTokenGenerationResponse> getServerPathF
     return ServerPathProvider.getServerUrlForTokenGeneration(endpointParams, httpClientProvider.anonymous(), port, appName)
       .thenApply(GetServerPathForTokenGenerationResponse::new);
   }
+
+  @Override
+  public void openHotspotInBrowser(OpenHotspotParams params) {
+    var hotspotId = params.getHotspotId();
+    var fileUri = create(params.getFileUri());
+    var folderForFileOptional = workspaceFoldersManager.findFolderForFile(fileUri);
+    if (folderForFileOptional.isEmpty()) {
+      lsLogOutput.error("Can't find workspace folder for file "
+        + fileUri.getPath() + " during attempt to open hotspot in browser.");
+      return;
+    }
+    var workspaceFolderUri = folderForFileOptional.get().getUri();
+    var branchNameOptional = branchManager.getReferenceBranchNameForFolder(workspaceFolderUri);
+    if (branchNameOptional.isEmpty()) {
+      lsLogOutput.error("Can't find branch for workspace folder "
+        + workspaceFolderUri.getPath() + " during attempt to open hotspot in browser.");
+      return;
+    }
+    var versionedIssue = securityHotspotsCache.get(fileUri).get(hotspotId);
+    var delegatingIssue = (DelegatingIssue) versionedIssue.getIssue();
+    var openHotspotInBrowserParams = new OpenHotspotInBrowserParams(workspaceFolderUri.toString(), branchNameOptional.get(), delegatingIssue.getServerIssueKey());
+    backendServiceFacade.getBackendService().openHotspotInBrowser(openHotspotInBrowserParams);
+  }
+
+  @Override
+  public void showHotspotRuleDescription(ShowHotspotRuleDescriptionParams params) {
+    var fileUri = params.fileUri;
+    var ruleKey = params.ruleKey;
+    var showHotspotCommandParams = new ExecuteCommandParams(SONARLINT_OPEN_RULE_DESCRIPTION_FROM_CODE_ACTION_COMMAND,
+      List.of(new JsonPrimitive(ruleKey), new JsonPrimitive(fileUri)));
+    CompletableFutures.computeAsync(cancelToken -> {
+      cancelToken.checkCanceled();
```

Comment:
For the cancelable functionality to work you need to return the future to the client